src/stores/TransactionStore.js

```diff
@@ -83,6 +103,51 @@ export default class TransactionStore {
     }
   };
 
+  createEvent = async ({
+    nbotMethods,
+    eventParams,
+    eventFactoryAddr,
+    escrowAmt,
+    gas,
+  }) => {
+    try {
+      // Construct params
+      const paramsHex = web3EthAbi.encodeParameters(
+        createEventFuncTypes,
+        eventParams,
+      ).substr(2);
+      const data = `0x${CREATE_EVENT_FUNC_SIG}${paramsHex}`;
+      // Send tx
+      const txid = await promisify(nbotMethods.transfer['address,uint256,bytes'].sendTransaction, [eventFactoryAddr, escrowAmt, data, { gas }]);
+      return txid;
+    } catch (err) {
+      throw err;
```

Comment:
need to handle this gracefully on the UI side. if we throw here, the entire dapp crashes. add `// TODO: show errror message` and show this error in an error dialog.